src/view/BoardView.java

```diff
@@ -232,8 +249,24 @@ public boolean isCastle(ChessButton clickedButton) {
     }
 
     public void propertyChange(PropertyChangeEvent e) {
-        previousMove.clear();
-        this.previousMove = null;
+            try {
+                CheckGameEndsState state = (CheckGameEndsState) e.getNewValue();
+                String color = "";
+                if (state.getiswhite()) {
+                    color = "white";
+                } else {
+                    color = "black";
+                }
+                if (state.getwin()) {
+                    JOptionPane.showMessageDialog(this, color + "win!");
```

Comment:
you should put a space next to "win!" -> " win!" so that the message will be "white win!" rather than "whitewin!" which it currently is